Services/CAS/classes/class.ilAuthProviderCAS.php

```diff
@@ -38,29 +67,54 @@ protected function getSettings(): ilCASSettings
     public function doAuthentication(ilAuthStatus $status): bool
     {
         $this->getLogger()->debug('Starting cas authentication attempt... ');
+        $baseUrl = $this->getIliasBaseUrl();
 
         try {
-            phpCAS::setDebug(false);
-            phpCAS::setVerbose(true);
+            // If you need a logger, uncomment this, the use statements
+            // and variables above and remove   phpCAS::setLogger(null);
+            // from below.
+            // The logger essentially gives you a trace through the
+            // entire CAS-call-stack.
+            //
+            // $this->logPath = tempnam(sys_get_temp_dir(), 'phpCAS');
+            // $this->logger = new Logger('name');
+            // $handler = new StreamHandler($this->logPath);
+            // $format = "%message%\n";
+            // $formatter = new LineFormatter($format);
+            // $handler->setFormatter($formatter);
+            // $this->logger->pushHandler($handler);
+            // phpCAS::setLogger($this->logger);
```

Comment:
I suggest injecting the PSR-compliant logger wrapped by the ILIAS logger, e.g.:

```php
$ilias_logger = $DIC->logger()->auth();

phpCAS::setLogger($ilias_logger->getLogger());
```

Reply:
or even better:

```php
phpCAS::setLogger($this->getLogger()->getLogger());
```

Reply:
@mjansenDatabay I suppose it should be only one `->getLogger()`.

What I noticed is that CAS will still log a lot with log level INFO; user details that might be unwanted in log files, so I will check if phpCAS would somehow allow to be less verbose by default.

```
[0dc17] [2024-11-19 19:27:15.474161] client_auth.INFO: phpCAS::trace:468 47AC .START (2024-11-19 19:27:15) phpCAS-1.6.1 ****************** [CAS.php:468]  
[0dc17] [2024-11-19 19:27:15.474320] client_auth.INFO: phpCAS::traceBegin:352 47AC .=> phpCAS::client('2.0', 'shib.rz.example.com', 443, 'idp/profile/cas', 'https://dev.ilias.example.com') [class.ilAuthProviderCAS.php:56]  
[0dc17] [2024-11-19 19:27:15.474460] client_auth.INFO: phpCAS::traceBegin:963 47AC .|    => CAS_Client::__construct('2.0', false, 'shib.rz.example.com', 443, 'idp/profile/cas', 'https://dev.ilias.example.com', true, CAS_Session_PhpSession) [CAS.php:369]  
[0dc17] [2024-11-19 19:27:15.474494] client_auth.INFO: phpCAS::trace:982 47AC .|    |    Session is not authenticated [Client.php:982]  
[0dc17] [2024-11-19 19:27:15.474528] client_auth.INFO: phpCAS::traceEnd:1080 47AC .|    <= ''  
[0dc17] [2024-11-19 19:27:15.474545] client_auth.INFO: phpCAS::traceEnd:375 47AC .<= ''  
[0dc17] [2024-11-19 19:27:15.474562] client_auth.INFO: phpCAS::traceBegin:1702 47AC .=> phpCAS::setNoCasServerValidation() [class.ilAuthProviderCAS.php:59]  
[0dc17] [2024-11-19 19:27:15.474581] client_auth.INFO: phpCAS::trace:1705 47AC .|    You have configured no validation of the legitimacy of the cas server. This is not recommended for production use. [CAS.php:1705]  
[0dc17] [2024-11-19 19:27:15.474596] client_auth.INFO: phpCAS::traceEnd:1707 47AC .<= ''  
[0dc17] [2024-11-19 19:27:15.474613] client_auth.INFO: phpCAS::traceBegin:1159 47AC .=> phpCAS::forceAuthentication() [class.ilAuthProviderCAS.php:61]  
[0dc17] [2024-11-19 19:27:15.474630] client_auth.INFO: phpCAS::traceBegin:1486 47AC .|    => CAS_Client::forceAuthentication() [CAS.php:1161]  
[0dc17] [2024-11-19 19:27:15.474646] client_auth.INFO: phpCAS::traceBegin:1592 47AC .|    |    => CAS_Client::isAuthenticated() [Client.php:1488]  
...
```

and hundreds of lines more. This is great for log level debug but not for log level info. So please do not yet merge.